### ISSUE TITLE
Has pincode?

### ISSUE THREAD

**Issue by mikecann** [fnzc/react-native-pincode#21]
I was just wondering how you would go about detecting if the user has actually set the pincode or not.

For for the following example:

1. User logs into the app and sets Pincode for quick access next time
2. User exits app
3. User enters app again. App needs to now determine if the user has set a pincode yet, in which case they should be directed to the "EnterPincode" screen, otherwise the user should be directed to the "LoginAndSetPincode" screen.

So the problem is around step 3, how does the app know which screen to send the user to? Obviously we could use localstorage to store if the user has set a pincode for the device but that means that react-native-pincode and the app are using two different stores for the same value so they could go out of sync? 

I guess one way to do it would be to use the "storePin" and "storedPin" properties to proxy access to  react-native-keychain but this seems somehow wasteful?

Thoughts?

**Comment by Haskkor** [comment]
Hi,

In our app flow, we ask the user to choose between a 'Continue' or a 'Start new' mode. So we never had to solve this problem.

If you let the package handle the storage of the pin, I don't see something better than the solution you already found. I'll give it a thought and come back to you.

Cheers,
Jeremy

**Comment by mikecann** [comment]
Hi Jeremy,

Thats a clever solution but I think we want to be able to determine for the user if they need to "continue" or to "start new" without asking them.

For now I have implemented storePin and storedPin manually and it seems to work but im not sure if I got all the biometric settings and the other settings right on the keychain side, so I may have to dig into the source some more and see what you have done here.

Thanks, feel free to close this if you feel there is no further action possible.

Mike
